SignalServiceKit/src/Messages/Interactions/OWSContact.m

```diff
@@ -141,28 +141,49 @@ @interface OWSContact ()
 
 @implementation OWSContact
 
+- (void)normalize
+{
+    self.phoneNumbers = [self.phoneNumbers
+        filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(OWSContactPhoneNumber *value,
+                                        NSDictionary<NSString *, id> *_Nullable bindings) {
+            return value.ows_isValid;
+        }]];
+    self.emails = [self.emails filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(OWSContactEmail *value,
+                                                               NSDictionary<NSString *, id> *_Nullable bindings) {
+        return value.ows_isValid;
+    }]];
+    self.addresses =
+        [self.addresses filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(OWSContactAddress *value,
+                                                        NSDictionary<NSString *, id> *_Nullable bindings) {
+            return value.ows_isValid;
+        }]];
+}
+
 - (BOOL)ows_isValid
 {
     if (self.displayName.ows_stripped.length) {
         return NO;
     }
-    // Contact is valid if at least one entry is valid.
+    BOOL hasValue = NO;
```

Comment:
Making this more strict upon reflect.

Reply:
Hmmm.... so if I have a contact entry for Bob which has 3 valid phone numbers and one invalid email, I can't share Bob's card?

I'd expect that I can share bob's card, but only the valid phone numbers are included.

TBH though, I'm not sure what we're getting from this validation logic. If I have something that our parser doesn't think is a valid phonenumber / address / email it seems like we'd still want to share whatever raw text the sender had, rather than just dropping it.